pkg/controllers/provisioning/provisioner.go

```diff
@@ -118,7 +118,9 @@ func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (resul
 	// with making any scheduling decision off of our state nodes. Otherwise, we have the potential to make
 	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
 	if !p.cluster.Synced(ctx) {
-		logging.FromContext(ctx).Debugf("waiting on cluster sync")
+		if time.Since(p.cluster.LastSyncTime()) > state.AbnormalClusterSyncDuration {
```

Comment:
I wonder if it's possible to encapsulate this behavior so that we don't actually have to handle logging it everywhere. What if it handled like the client-go "waiting 1.5s due to client-side throttling" error message. If we keep calling cluster synced and we stay false for a long enough time-frame, the method just logs that it hasn't synced itself

Reply:
> If we keep calling cluster synced and we stay false for a long enough time-frame, the method just logs that it hasn't synced itself

Yeah I was thinking about this too. Just in cluster state we log in the Synced() call if it hasn't synced. This way we also don't have to expose "lastSyncedTime" as well.

Reply:
Moved it into the Synced() method and am using a change monitor lmk what you think.

Reply:
https://github.com/kubernetes/client-go/blob/aa7909e7d7c0661792ba21b9e882f3cd6ad0ce53/rest/request.go#L625 is how they log this in client-go for clientside throttling. 

@jonathan-innis  did you mean just copy the error message style? "waiting for "x seconds" dur to cluster state being out of sync?